components/quick_input.jsx

```diff
@@ -73,18 +88,59 @@ export default class QuickInput extends React.PureComponent {
         this.input = input;
     }
 
+    handleClear = () => {
+        if (this.props.handleClear) {
+            this.props.handleClear();
+        }
+
+        this.value = '';
+    }
+
     render() {
-        const {value, inputComponent, ...props} = this.props;
+        const clearableTooltip = (
+            <Tooltip id={'InputClearTooltip'}>
+                <FormattedMessage
+                    id={'input.clear'}
+                    defaultMessage='Clear input'
+                />
+            </Tooltip>
+        );
+
+        const {value, inputComponent, clearable, ...props} = this.props;
 
         Reflect.deleteProperty(props, 'delayInputUpdate');
+        Reflect.deleteProperty(props, 'handleClear');
 
-        return React.createElement(
+        const inputElement = React.createElement(
             inputComponent || 'input',
             {
                 ...props,
                 ref: this.setInput,
                 defaultValue: value, // Only set the defaultValue since the real one will be updated using componentDidUpdate
             }
         );
+
+        return (<div>
+            {inputElement}
+            {clearable && this.input && this.value && this.value !== '' &&
```

Comment:
Could we apply a trim here to cover all empty spaces ?
```suggestion
            {clearable && this.input && this.value && this.value.trim() !== '' &&
```

Reply:
do we not want the `X` to show up when spaces are there? 0/5 either way.

Reply:
The behavior so far has been to not show the `x` when there is only spaces, in [`search_bar`](https://github.com/mattermost/mattermost-webapp/blob/65c23960d0ae5c6cd175560ae9c8318bd5966b96/components/search_bar/search_bar.jsx#L214) and [`marketplace_modal`](https://github.com/mattermost/mattermost-webapp/blob/65c23960d0ae5c6cd175560ae9c8318bd5966b96/components/plugin_marketplace/marketplace_modal.js#L215)

But not on `admin_sidebar` 🤔 where it does show the `x` where there is only spaces. 

@asaadmahmood from the UX perspective, what should be the correct behavior?

Reply:
I will wait for @asaadmahmood  comment to fix this.

Reply:
@asaadmahmood is out, @michaelgamble can you give use your UX opinion here?

Reply:
@larkox Based on UX feedback [here](https://community-release.mattermost.com/core/pl/wj43nryza7gi8b8rz81dqpzjgw), the `x` should appear even if it only has white spaces so we should not include the `.trim()` here